libr/core/agraph.c

```diff
@@ -4388,21 +4395,22 @@ R_API int r_core_visual_graph(RCore *core, RAGraph *g, RAnalFunction *_fcn, int
 	core->cons->event_resize = (RConsEvent) agraph_refresh_oneshot;
 
 	r_cons_break_push (NULL, NULL);
-#if 0
-	// XXX wrong usage or buggy RGraph.domTree()
-	// dominance tree here
-	const RList *l = r_graph_get_nodes (g->graph);
-	RGraphNode *root = r_list_first (l);
-	if (root) {
-		RGraph *dg = r_graph_dom_tree (g->graph, root);
-		if (dg) {
-			g->graph = dg;
-		} else {
-			R_LOG_WARN ("Cannot compute the dominance tree");
-			sleep (1);
+	if (mode == 3) { // XXX wrong usage or buggy RGraph.domTree()
+		// dominance tree here
+		const RList *l = r_graph_get_nodes (g->graph);
+		RGraphNode *root = r_list_first (l);
+		if (root) {
+			RGraph *dg = r_graph_dom_tree (g->graph, root);
+			if (dg) {
+				R_LOG_INFO ("pwn");
+				r_graph_free (g->graph);
```

Comment:
this likely causes segfaults, because the domtree nodes do not copy from the original graph, but point to the nodes from the original graph

Reply:
better leak than double free